src/CoreWCF.Http/tests/Helpers/ServiceHelper.cs

```diff
@@ -302,6 +305,18 @@ public static Stream GetStreamWithStringBytes(string s)
             return stream;
         }
 
+        public static Stream GetAsyncStreamWithStringBytes(string s)
+        {
+            Stream inner = new MemoryStream();
+            PopulateStreamWithStringBytes(inner, s);
+#if NETFRAMEWORK
```

Comment:
I prefer to avoid conditional compilation whenever possible. It has a habit of breaking refactoring tooling and makes it harder to see coding mistakes in the IDE before doing a full multi-targeted build. Can you switch to a runtime check similar to what you did in XmlDictionaryWriterExtensions? It doesn't need to be cached in a static, you can just look it up each time to keep the code simpler.